### PR TITLE
Connect Device Account Step: Remove polling and set providerID

### PR DESCRIPTION
**Pull request by bxh**
## Overview

This PR removes polling as it turns out to be unnecessary. Also set `providerID` based on `deviceType` and host name.

## Security

REMINDER: All file contents are public.

- [x] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc.
  - [x] Please verify that you double checked that .storybook/preview.js does not contain your participant access key details. 
  - [x] There are no temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
- [x] These changes do not introduce any security risks, or any such risks have been properly mitigated.

Describe briefly what security risks you considered, why they don't apply, or how they've been mitigated.

In the case of misconfigured providerID, a participant will just be shown a wrong yet secure provider login website.

### Testing
- [ ] MyDataHelps iOS app
- [ ] MyDataHelps Android app
- [x] [MyDataHelps website](mydatahelps.org)

### Documentation
- [ ] I have added relevant Storybook updates to this PR as well.
- [ ] If this feature requires a developer doc update, tag @CareEvolution/api-docs.

Consider "Squash and merge" as needed to keep the commit history reasonable on `main`.

## Reviewers

Assign to the appropriate reviewer(s). Minimally, a second set of eyes is needed ensure no non-public information is published. Consider also including:
- Subject-matter experts
- Style/editing reviewers
- Others requested by the content owner

### CODE REVIEW

File: src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ export default function () {
 
     useEffect(() => {
         // Start polling for connected status after a providerID is selected.
-        if (!providerID || connected) return;
+        if (!providerID || connected || !MyDataHelps.token) return;
```
Comment (ashleyrtate) [inline src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:39]: Not sure I understand why this is needed. Doesn't getExternalAccounts() set the token?

Comment (chrisnowak) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:39]: But yeah, as @ashleyrtate pointed out, I think the token actually would get set by the first call to getExternalAccounts() (because that is the first API request).

Actually, backing up, what happens if they visit this step and they've already connected Fitbit?  (I can't test it in QA right now since it seems to be complaining about missing providerID)

Comment (chrisnowak) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:39]: We probably want to check if they have already connected the device and auto-proceed if so; so I think when this step loads you would need to call getExternalAccounts immediately...

Comment (bxh) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:39]: 1. A _MyDataHelpsDesigner_ user add a ConnectDeviceAccountStep to a survey.
2. They set a providerID.
3. At this time, the rendered ConnectDeviceAccountStep step starts polling, which always fail due to lack of token.

This PR aims to address this issue.

Comment (ashleyrtate) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:39]: I would set a "designMode" flag or something to make that explicit.

Comment (bxh) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:39]: @chrisnowak @ashleyrtate ^

Comment (chrisnowak) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:39]: I thought it should only start polling after the button is clicked?

Comment (bxh) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx]: No, this is the ConnectDeviceAccountStep.

> When the user visits this step it would check to see if they have the account connected; if so it would automatically proceed.

https://github.com/CareEvolution/Consumers/issues/11603

Comment (chrisnowak) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx]: Ah sorry I didn't notice this in the original PR - there was another issue I also was wondering about the other day:
- Seems like we should only check once when the step is loaded vs. always poll, right?
- Also...curious which / what platform is forcing us to poll anyways?  ConnectExternalAccount should return a promise so you should know when it is complete, no?

Comment (bxh) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx]: Polling is needed as we wait for the participant to connect to the device account. But you are right that this process can start **after** the participant clicks the button. In the initialization phase, we can just make one request.

Comment (bxh) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx]: (I'm working on the change)

Comment (chrisnowak) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx]: @bxh why is it needed?  We get a promise back when they click the button that resolves when they connect the account

Comment (bxh) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx]: Clicking the button only invokes `.connectExternalAccount`. That does not mean the participant is instantaneously connected, right? They still need to proceed to the third party website to sign in themselves.

Comment (chrisnowak) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx]: Right...after they do the connection the promise returned by connectExternalAccount resolves with a status (or fails if they cancel I think)

Comment (chrisnowak) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx]: e.g. you should be able to do something like
```
MyDataHelps.connectExternalAccount(1).then(() => {
   //stuff that should happen after connection is complete
});
```

Comment (bxh) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx]: Oh, I didn't realize that `connectExternalAccount` resolves _after_ a connection is complete.

**Comment by bxh** [comment]
@chrisnowak I've refactored ConnectDeviceAccountStep based on your suggestion.

1. Removed polling. 
2. Moved all `MyDataHelps` stuff to the container component.

A follow up question, how should we handle errors when `connectExternalAccount` fails?

**Comment by bxh** [comment]
On a related note, I'll remove polling from ConnectEhrStep using another PR.

File: src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx [hunk 2]
```diff
@@ -2,13 +2,26 @@ import React, { useState, useEffect } from "react";
...
-    const [providerID, setProviderID] = useState<number>();
-    const [connected, setConnected] = useState<boolean>();
+    const [deviceType, setDeviceType] = useState<string>(props.deviceType ?? "");
+    const [providerID, setProviderID] = useState<number>(props.providerID ?? 0);
+
+    function onConnect() {
+        MyDataHelps.connectExternalAccount(providerID, {
+            openNewWindow: true,
+        }).then(function() {
+            console.log("Completing step. Connected to provider ID", providerID);
```
Comment (chrisnowak) [inline src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:21]: Remove console log before merging

File: src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx [hunk 3]
```diff
@@ -2,13 +2,26 @@ import React, { useState, useEffect } from "react";
...
+            MyDataHelps.completeStep("")
```
Comment (chrisnowak) [inline src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:22]: Maybe put the providerID in here

File: src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx [hunk 4]
```diff
@@ -28,44 +41,36 @@ export default function () {
...
-            const accounts = await MyDataHelps.getExternalAccounts();
-            const containProvider = accounts.some(
-                (acc) => acc.provider.id === providerID
-            );
-            if (!connected && containProvider) {
-                setConnected(containProvider);
+        MyDataHelps.getExternalAccounts().then(function(accounts) {
+            const connected = accounts.some((acc) => acc.provider.id === providerID);
+            if (connected) {
+                console.log("Completing step. Already connected to provider ID", providerID);
```
Comment (chrisnowak) [inline src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:58]: remove before merging

File: src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx [hunk 5]
```diff
@@ -28,44 +41,36 @@ export default function () {
...
+                MyDataHelps.completeStep("");
```
Comment (chrisnowak) [inline src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:59]: Would put providerID in here actually

File: src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx [hunk 6]
```diff
@@ -2,13 +2,26 @@ import React, { useState, useEffect } from "react";
...
     const [title, setTitle] = useState<string>();
     const [text, setText] = useState<string>();
     const [styles, setStyles] = useState<any>({});
-    const [deviceType, setDeviceType] = useState<string>("");
-    const [providerID, setProviderID] = useState<number>();
-    const [connected, setConnected] = useState<boolean>();
+    const [deviceType, setDeviceType] = useState<string>(props.deviceType ?? "");
+    const [providerID, setProviderID] = useState<number>(props.providerID ?? 0);
+
+    function onConnect() {
```
Comment (chrisnowak) [inline src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:17]: Why is this moved to the container?  

I think the idea with the container is that it should just be thin (no stories) and just get the config / styles and render the step...ultimately this pattern is a bit wonky though and I am regretting it in favor of having some sort of "previewState" or "designerMode" flag like the others.  But...maybe neither here nor there for now

**Comment by chrisnowak** [comment]
Seems reasonable other than the comments I made...I think most of my issues are around the container pattern which is probably due for a refactor anyways.  🚢

**Comment by chrisnowak** [comment]
Oh, but same comment as on other PR, I would make sure to test the connectExternalAccount promise flows on iOS/Android/Web to ensure it works consistently for the various states.  Obviously if the user cancels the connection nothing should happen so want to make sure we aren't proceeding in that case with the code as written or if there is some special handling it requires to detect that case

File: src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx (hunk 6 above)
Comment (bxh) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:17]: I was under the impression that the containers are for behavior, whereas the UI components are just for presentation.

**Comment by bxh** [comment]
@chrisnowak Can you approve the pull request?

File: src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx [hunk 7]
```diff
@@ -2,13 +2,34 @@ import React, { useState, useEffect } from "react";
...
+        MyDataHelps.connectExternalAccount(providerID, {
+            openNewWindow: true,
+        }).then(function () {
+            MyDataHelps.completeStep(providerID);
+        });
+    }
+
+    function isProd() {
+        return (
+            window.location.host.endsWith(".com") ||
```
Comment (chrisnowak) [inline src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:29]: This isn't the right logic; you need to check MyDataHelps.baseUrl like we do here: https://github.com/CareEvolution/MyDataHelpsUI/blob/main/src/components/container/ConnectFitbit/ConnectFitbit.tsx#L18

Note that MyDataHelps.baseUrl won't be populated until you have a token - so you should probably invoke getExternalAccounts first

File: src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx [hunk 8]
```diff
@@ -2,13 +2,34 @@ import React, { useState, useEffect } from "react";
...
     const [title, setTitle] = useState<string>();
     const [text, setText] = useState<string>();
     const [styles, setStyles] = useState<any>({});
-    const [deviceType, setDeviceType] = useState<string>("");
-    const [providerID, setProviderID] = useState<number>();
-    const [connected, setConnected] = useState<boolean>();
+    const [deviceType, setDeviceType] = useState<string>(
+        props.deviceType ?? ""
+    );
+    const [providerID, setProviderID] = useState<number>(props.providerID ?? 0);
```
Comment (chrisnowak) [inline src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:17]: Would drop the state param and prop since it's not needed, you can just get the providerID with a method

File: src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx [hunk 9]
```diff
@@ -25,47 +46,57 @@ export default function () {
...
+                            config.properties.deviceType
+                    );
             }
-            setProviderID(config.properties.providerID);
 
             setStyles(config.styles ?? {});
         });
     }, []);
 
     useEffect(() => {
```
Comment (chrisnowak) [inline src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:76]: These separate useEffects aren't necessary anymore; would just chain getStepConfiguration() into getExternalAccounts()

File: src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx (hunk 7 above)
Comment (bxh) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx]: I'm including both https://mdhorg.ce.dev and https://mydatahelps.dev, since both of them are in use.

File: src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx [hunk 10]
```diff
@@ -2,20 +2,75 @@ import React, { useState, useEffect } from "react";
...
+
+    function isTest() {
+        return (
+            !MyDataHelps.baseUrl ||
+            MyDataHelps.baseUrl.startsWith("https://mdhorg.ce.dev") ||
+            MyDataHelps.baseUrl.startsWith("https://mydatahelps.dev")
+        );
+    }
+
+    function convertToProviderID(deviceType: string) {
```
Comment (chrisnowak) [inline src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:35]: Will probably refactor this to use the new methods in https://github.com/CareEvolution/MyDataHelpsUI/pull/118 so this can be centralized

**Review by chrisnowak — APPROVED** [review]

File: src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx (hunk 10 above)
Comment (bxh) [reply src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx:28]: Fixed.
